pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -146,8 +146,8 @@ func resolveResultRef(pipelineState PipelineRunState, resultRef *v1beta1.ResultR
 	if referencedPipelineTask == nil {
 		return nil, resultRef.PipelineTask, fmt.Errorf("could not find task %q referenced by result", resultRef.PipelineTask)
 	}
-	if !referencedPipelineTask.isSuccessful() {
-		return nil, resultRef.PipelineTask, fmt.Errorf("task %q referenced by result was not successful", referencedPipelineTask.PipelineTask.Name)
+	if !referencedPipelineTask.isSuccessful() && !referencedPipelineTask.isFailure() {
+		return nil, resultRef.PipelineTask, fmt.Errorf("task %q referenced by result was not successful and not failure", referencedPipelineTask.PipelineTask.Name)
```

Comment:
What other status could it be ? Maybe the text should say "not finished (success or failure)", or something like that ?

Reply:
> What other status could it be ? Maybe the text should say "not finished (success or failure)", or something like that ?

Good suggestion, I have adjusted. Thanks.